File: mcorg.backends/project-service/src/main/java/app/mcorg/project/domain/model/project/ProjectDependency.java
package app.mcorg.project.domain.model.project;

import app.mcorg.common.domain.model.Priority;

public record ProjectDependency(String projectId,
                                Priority priority,
                                Direction direction) {
    public enum Direction {
        DEPENDS_ON,
        IS_DEPENDED_ON_BY
    }
}

File: webapp/src/main/kotlin/app/mcorg/domain/model/project/ProjectDependency.kt
package app.mcorg.domain.model.project

data class ProjectDependency(
    val dependentId: Int,
    val dependentName: String,
    val dependentStage: ProjectStage,
    val dependencyId: Int,
    val dependencyName: String,
    val dependencyStage: ProjectStage
)
